### PR TITLE
Game logic

### PR DESCRIPTION
**Pull request by rOluochKe**

### CODE REVIEW

File: bin/main.rb [hunk 1]
```diff
@@ -1,49 +1,26 @@
...
 end
 
-#run game
-ttt = Board.new
-player = ["x", "y"]
+# run game
+ttt = Main.new
+player = %w[x y]
 
 ttt.show_instructions
```
Comment (AdnanAfsari) [inline bin/main.rb]: Its better to keep everything within the class.

File: lib/board.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# rubocop: disable Style/FrozenStringLiteralComment, Lint/MissingCopEnableDirective
+
+module Board
+  def show
```
Comment (AdnanAfsari) [inline lib/board.rb]: User interaction commands should be in one executable file main.rb

File: lib/board.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    puts @top_row
+    puts @mid_row
+    puts @bot_row
+  end
+
+  def show_instructions
```
Comment (AdnanAfsari) [inline lib/board.rb]: User interaction commands should be in one executable file main.rb

File: lib/player.rb [hunk 4]
```diff
@@ -0,0 +1,56 @@
+# rubocop: disable Style/FrozenStringLiteralComment, Lint/MissingCopEnableDirective
+
+module Player
+  # rubocop: disable Naming/UncommunicativeMethodParamName
```
Comment (AdnanAfsari) [inline lib/player.rb]: Its not a good idea disabling linter, you better fix those errors to maintain standard code 😸

**Review by AdnanAfsari — CHANGES_REQUESTED** [review]
Hello @rOluochKe @AndresFMoya ✋ 

Nice try guys but you have to make some changes and improve your game.Please check my comments on those files and below and submit for another review after making changes.

- [x] I was trying to play the game in my local machine and i got this error.please make sure its working.
Screenshot 2019-09-12 at 1 56 02 AM

- [x] we can make our README.md more descriptive for example adding instructions how to play the game etc, you can check other project for inspiration.

Note: Please  add a short summary of changes in Pull Request description 😄.

Happy Coding

File: lib/player.rb (hunk 4 above)
Comment (AndresFMoya) [reply lib/player.rb]: Hello Adnan. Thanks for your review. Now We don't know how to solve the cyclomatic and perceived complexity errors. Could you please point us the right address. Thanks!

**Comment by Oghenebrume50** [comment]
reviewing...

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
`Welcome to terminal tic tac toe!
The following are the move positions you are able to make:
|tl|tc|tr|
|ml|mc|mr|
|bl|bc|br|

It's Player x's turn.
j
| | | |
| | | |
| | | |
It's Player y's turn.
r
| | | |
| | | |
| | | |
It's Player x's turn.
3
| | | |
| | | |
| | | |
It's Player y's turn.
4
| | | |
| | | |
| | | |
It's Player x's turn.
5
| | | |
| | | |
| | | |
It's Player y's turn.
2
| | | |
| | | |
| | | |
It's Player x's turn.
1
| | | |
| | | |
| | | |
It's Player y's turn.
9
| | | |
| | | |
| | | |
It's Player x's turn.
8
| | | |
| | | |
| | | |
It's Player y's turn.
0
| | | |
| | | |
| | | |
It's Player x's turn.
-
| | | |
| | | |
| | | |
It's Player y's turn.
2
| | | |
| | | |
| | | |
It's Player x's turn.
4
| | | |
| | | |
| | | |
It's Player y's turn.
5
| | | |
| | | |
| | | |
It's Player x's turn.
6
| | | |
| | | |
| | | |
It's Player y's turn.
7
| | | |
| | | |
| | | |
It's Player x's turn.
8
| | | |
| | | |
| | | |
It's Player y's turn.
5
| | | |
| | | |
| | | |
It's Player x's turn.
6
| | | |
| | | |
| | | |
It's Player y's turn.
7
| | | |
| | | |
| | | |
It's Player x's turn.
8
| | | |
| | | |
| | | |
It's Player y's turn.
9
| | | |
| | | |
| | | |
It's Player x's turn.
6
| | | |
| | | |
| | | |
It's Player y's turn.
4
| | | |
| | | |
| | | |
It's Player x's turn.
1
| | | |
| | | |
| | | |
It's Player y's turn.
2
| | | |
| | | |
| | | |
It's Player x's turn.
3
| | | |
| | | |
| | | |
It's Player y's turn.
4
| | | |
| | | |
| | | |
It's Player x's turn.
5
| | | |
| | | |
| | | |
It's Player y's turn.
7
| | | |
| | | |
| | | |
It's Player x's turn.
8
| | | |
| | | |
| | | |
It's Player y's turn.
9
| | | |
| | | |
| | | |
It's Player x's turn.
tl
|x| | |
| | | |
| | | |
It's Player y's turn.
tm
|x| | |
| | | |
| | | |
It's Player x's turn.
tc
|x|x| |
| | | |
| | | |
It's Player y's turn.
tr
|x|x|y|
| | | |
| | | |
It's Player x's turn.
bm
|x|x|y|
| | | |
| | | |
It's Player y's turn.
bc
|x|x|y|
| | | |
| |y| |
It's Player x's turn.
bl
|x|x|y|
| | | |
|x|y| |
It's Player y's turn.
br
|x|x|y|
| | | |
|x|y|y|
It's Player x's turn.
mc
|x|x|y|
| |x| |
|x|y|y|
It's Player y's turn.
ml
|x|x|y|
|y|x| |
|x|y|y|
It's Player x's turn.
mr
|x|x|y|
|y|x|x|
|x|y|y|
It's Player y's turn.
tm
|x|x|y|
|y|x|x|
|x|y|y|`

This is an output from an attempted play of the game, I believe we can make it more user-friendly, by outputting a warning on wrong input, and as you can see the game did not end properly, after a draw.

[NOTE]: In OOP generally, modules are for objects that you need to use once, for example the Board can be a module but not the player because the player is needed more than once, we need two instances of a player

[IMPORTANT] please separate the logic from the display code, things like puts, print, chomp should be in one final and only that file
this is an OOP project so we want to use OOP concepts and methodology. 
Each object should be in it's own class, for example,  player, board and game  could be in seperate classes. 

Think about atributes and behaviours, what are the behaviours of an object, these are the likely methods needed, the attributes are the characteristics, in a human it would be things like hand, leg, eye. In a Player it could be name, symbol, etc. 

Read about single responsibility principle and DRY

File: lib/player.rb [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    return true if (0..2).all? { |i| check(1, i, player) }
+    return true if (0..2).all? { |i| check(2, i, player) }
+    # vertical wins
+    return true if (0..2).all? { |i| check(i, 0, player) }
+    return true if (0..2).all? { |i| check(i, 1, player) }
+    return true if (0..2).all? { |i| check(i, 2, player) }
+    # diagonal wins
+    return true if (0..2).all? { |i| check(i, i, player) }
+    return true if (0..2).all? { |i| check(i, 2 - i, player) }
+  end
```
Comment (Oghenebrume50) [inline lib/player.rb:18]: read about DRY, and as a note on single responsibility, it is not the job of the player to determine who won the game, the game should be smart enough to do that :smile:

**Review by Oghenebrume50 — CHANGES_REQUESTED** [review]
**Please fix these things and request for a re-review** :pray: 

oh and do not forget to fix linter issues too

**Review by maelfosso — CHANGES_REQUESTED** [review]
Hello Coders,

# Really good job :+1:  It's was a pleasure to review it.

Everything is good but only one thing to change: **No puts, gets.chomp and print** into `lib/`

Kindly, delete all the **puts, gets.chomp and print** into classes `lib/player.rb` and `lib/game.rb`
Then re-submit a code review request
Thanks

**Comment by Ispirett** [comment]
# Reviewing

File: bin/main.rb [hunk 6]
```diff
@@ -1,59 +1,55 @@
...
-  end
-
-  def check(a, b, player)
-    @grid[a][b] == player
-  end
-
-  def move(position, player)
-    puts "Pick a number to play"
+require_relative '../lib/game'
+class Playgame
```
Comment (Ispirett) [inline bin/main.rb:4]: Convention for a class name is every new word should begin with a capital letter eg: `PlayGame`

**Review by Ispirett — APPROVED** [review]
Very Good job Folks its a thumbs up from me :+1: 
image

I suggest adding a more descriptive readme!.

**Pull request merged**